Project-2/READMEJvonderh.md

```diff
@@ -25,18 +25,29 @@ Tip #4: Work together as a team.
 
 2. Use your axioms from question 1 as the basis of an ALCI T-Box. Supplement this T-box with whatever other axioms you like, as well as an A-box, so that you ultimately construct a knowledge base K = (T,A). Provide a _model_ of K. This may be graphical or symbolic or both. 
 PP= ∀part_of
+
 ipp= ∀part_of
+
 ip= ∀part_of-
+
 O= ∀part_of^∃part_of
+
 D= ∀~part_of
 
 Lazarus: PP, x
+
 (Lazarus, City)
+
 City: Ipp, Y
+
 Lazarus: Part_of City
+
 City: Part_of Lazarus
+
 Proper part: x, y
+
 overlaps: x, y
+
```

Comment:
Q3. There is no “^” in DL, you should use “⊓” instead.
(a) ∀R.(∃R.^∀R.)
This means: for all x, each of x’s all R-successors has some R-successor is R-related to all things. This is what (a) means.

(b) ∃R.(∀R.^∃R.)
This reads in natural language: for all things (y?), each has some R-filler (z?) which is R-related to everything (y? wrong) and has some R-filler. This is not what (b) means.

(c) If you really wanna translate (setting the issue with the unbounded x), then it should be “∀R.∃R.∀R.A”

(d) If you really wanna translate (setting the issue with the unbounded x), it should be “(∀R.A) ⊓ (∃R.B)”.

Q4. Here we wanna use a formula to differ between two models in the sense that one model satisfies that formula while another model does not satisfy it. So you can save one formula provided here and delete another one. And then you should provide a graph using Mermaid.

Q5. A simple pair of models are: M1 includes two points such that a is r-related to b, M2 includes 2 points such that c is R-related to d. These two models cannot be distinguished by (≥1 R).

Q6. Note that in DL all quantified formulas include an implicit universal quantifier, so each quantified formula in DL must pick out “all individuals satisfying some complex condition”. For example, “∃r.C” does not mean “there is something such that it is an instance of the concept r” (a typo here: the concept is C but not r); rather, it means that “for all things x, x is R-related to some y, which is an instance of C”. Also, r is a binary relation, so you should not say something like “… is an instance of r”, but should say “… is r-related to …” or “… has a r-filler which is …”. Similar problems generally exist in (a)~(d)

Similarly, “∃r-.C” should mean that “for all things x, x is R-inversed-related to some y, which is an instance of C”. 

(c) “nr” is not a single concept, “n” and “r” should be treated separately so *n* refers to the number of r-successors of something. “≤n” means that the number of something’s r-successors is less than or equal to *n*.

Q7. For finite model property, an important feature is that the interpretation has a finite domain set and this is why a formula can be check to be true or false under that interpretation in finite moves.

I don’t follow what you mean by “not known for sure”. Do you refer to the open world assumption that any thing which is not explicitly asserted is open to some property?

When you say “This would mean that this sentence can be understood as false within this system, as being a bachelor does not prevent someone from having had a kid.” I don’t see why this shows the finite model property, or it can be done in finite steps.

“If a system lacks that property, then there may be a sentence and interpretation which may not be true or false in a way we can determine.” – I guess an infinite but cyclic model can also show whether some sentence is true or false, so the usefulness of finite model property should be more concerned with **efficiency (of mechanic inference)**. Sometimes we are not interested in finite model property because we are willing to sacrifice it for a stronger expressive power. For example, ALCIN lacks finite model property.

Q8. Tree model property would be especially useful in consistency check of knowledge base or domain ontology. A domain ontology is established by following Aristotelian definitional schema, so it naturally has tree model property.

Q9. If cOa is inferred, then it should be highlighted in yellow. So it seems that you directly presume but not infer cOa?

Also, your explanation of tableau algorithm sounds good! But I am not sure how your explanation shows a tableau, or a tree-like inferential structure. But according to Theorem 2.17 (Baader, p. 32), we can transform an inference problem to an inconsistency check problem, and then we can directly the tableau algorithm “Expand” mentioned in Baader, Chapter 4. 

Q10. Your graph does not show (c) aPb, bPf, and aPf, and (d) aDg, and it seems that your inference results have some problem as the little red triangle in the right-down corner.

Reply:
3, 4, and 5 should now be complete and resolved. 6 should also be corrected.

For 7, I have added a point noting the finite domain size. For the "not known for sure" I am referring to the statements not being theorems of the system (I know it is a simplified version of "theorem" but that should be sufficient for a lay person).

For the false sentence point, this is partially a result of being very simplified (as far as I am concerned I have failed to meet the prompt the moment a logical symbol appears), but the idea there is in an intuitive interpretation (made regarding the words in the system) can result in the sentence in question being false in an intuitive sense. As such, an interpretation of the words "bachelor" and "parent" can falsify the sentence in question if neither precludes one having had a child as a bachelor.

I have added clauses or sentences noting those consequences you mentioned for 7 and 8. Was hard pressed for direct uses so that helps.

For 9 I am not sure why the yellow outline disappeared. It is not asserted in that ontology and is a result of the reasoner but for some reason stopped showing in yellow when I ran the reasoner at the time of that screenshot.

For the tableau reasoning there, I am not sure I need to show a tree-like structure from that prompt. All it asks is for me to explain how the system is reaching that conclusion, and it does not ask why it is a specifically tableau algorithm. If I missed something let me know, but I don't think a change is merited there since the prompt does not ask for the information you mentioned.

For 10 the picture has been corrected. Since Beverly has said it is fine to just have one screenshot I am still just using a screenshot of a since the only individual outside of a which would require a screenshot is b.

Reply:
If the screenshots have inexplicably stopped working for you lmk and I can just send them. They should be the same for the first and include the missing a-involved relationships for the second.

Reply:
Images should be fixed